src/stream/src/common/log_store_impl/kv_log_store/mod.rs

```diff
@@ -262,6 +262,50 @@ mod v1 {
     });
 }
 
+pub(crate) use v2::KV_LOG_STORE_V2_INFO;
+
+/// A new version of log store schema. Compared to v1, the v2 added a new vnode column to the log store pk,
+/// becomes `epoch`, `seq_id` and `vnode`. In this way, providing a log store pk, we can get exactly one single row.
+///
+/// In v1, dist key is not in pk, and we will get an error in batch query when we try to compute dist key in pk indices.
+/// Now in v2, since we add a vnode column in pk, we can set the vnode index in pk correctly, and the batch query can be
+/// correctly executed. See <https://github.com/risingwavelabs/risingwave/issues/14503> for details.
```

Comment:
Previously the panics comes from computing the `dist_key_in_pk_indices` field of `StorageTableDesc` for batch query. It will panic or return error when the dist key is not in the pk indices. 

But after https://github.com/risingwavelabs/risingwave/pull/14570 it will prefer using the vnode column instead of pk indices if there exists vnode column in the table catalog. So in this PR we added a vnode column and it will not generate the  `dist_key_in_pk_indices` and there will not be any error.